### ISSUE TITLE
anaconda-client broken on 3.12 ?

### ISSUE THREAD

**Issue by Carreau** [Anaconda-Platform/anaconda-client#689]
It depends on Clyient, that has not been updated in 5+ years, and import `imp` that was deprecated, and removed in 3.12.
```
Traceback (most recent call last):
  File "/opt/conda/bin/anaconda", line 6, in <module>
    from binstar_client.scripts.cli import main
  File "/opt/conda/lib/python3.12/site-packages/binstar_client/__init__.py", line 19, in <module>
    from . import errors
  File "/opt/conda/lib/python3.12/site-packages/binstar_client/errors.py", line 3, in <module>
    from clyent.errors import ClyentError
  File "/opt/conda/lib/python3.12/site-packages/clyent/__init__.py", line 5, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
```

See https://github.com/scientific-python/upload-nightly-action/issues/43

**Comment by mbargull** [comment]
@Carreau, this has been fixed with the new `1.12.2` release (which removed the `clyent` dependency).
That package should already be available on Anaconda's channel and will be available on `conda-forge` soon (but even `1.12.1` is now usable with Python 3.12 on `conda-forge` because of https://github.com/conda-forge/clyent-feedstock/pull/6 .)

**Comment by matthewfeickert** [comment]
For those waiting for conda-forge deployment, follow https://github.com/conda-forge/anaconda-client-feedstock/pull/50.
